Sources/DistributedActors/Cluster/Cluster+Event.swift

```diff
@@ -19,11 +19,35 @@ extension Cluster {
     /// Represents cluster events, most notably regarding membership and reachability of other members of the cluster.
     ///
     /// Inspect them directly, or `apply` to a `Membership` copy in order to be able to react to membership state of the cluster.
-    public enum Event: Codable, Equatable {
-        case snapshot(Membership)
-        case membershipChange(MembershipChange)
-        case reachabilityChange(ReachabilityChange)
-        case leadershipChange(LeadershipChange)
+    public struct Event: Codable, Equatable {
+        internal enum _Event: Equatable {
+            case snapshot(Membership)
+            case membershipChange(MembershipChange)
+            case reachabilityChange(ReachabilityChange)
+            case leadershipChange(LeadershipChange)
+        }
+
+        internal let underlying: _Event
+
+        private init(_ underlying: _Event) {
+            self.underlying = underlying
+        }
+
+        public static func snapshot(_ membership: Membership) -> Event {
+            .init(.snapshot(membership))
+        }
+
+        public static func membershipChange(_ membershipChange: MembershipChange) -> Event {
+            .init(.membershipChange(membershipChange))
+        }
+
+        public static func reachabilityChange(_ reachabilityChange: ReachabilityChange) -> Event {
+            .init(.reachabilityChange(reachabilityChange))
+        }
+
+        public static func leadershipChange(_ leadershipChange: LeadershipChange) -> Event {
+            .init(.leadershipChange(leadershipChange))
+        }
```

Comment:
Hmmm... so this one I'd kind of want to keep as an enum to be honest :\

The primary way to use it is to switch over it in an async for loop of events...

Perhaps we should do this `case _PLEASE_DO_NOT_EXHAUSTIVELY_MATCH_THIS_ENUM_NEW_CASES_MIGHT_BE_ADDED_IN_THE_FUTURE` trick for this one specifically... other SSWG libs do this for such cases.

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/1025/commits/7ea7b1f46316ddbaff5d6861cbd6799bd9b80e02

Internally we just log error and/or ignore the "fake" event. Is that ok? I don't think we should do `default` because we will forget to update the `switch` when we add new event in the future. It doesn't feel right to throw error or crash either.

Reply:
Agreed, that looks good